synapse/config/server.py

```diff
@@ -88,6 +91,19 @@ def read_config(self, config):
             "restrict_public_rooms_to_local_users", False,
         )
 
+        self.default_room_version = config.get(
+            "default_room_version", DEFAULT_ROOM_VERSION,
+        )
+
+        if self.default_room_version not in KNOWN_ROOM_VERSIONS:
```

Comment:
people will forget to put quotes around the field in the yaml. can you convert to `str` before doing the lookup?